mods/default/item_entity.lua

```diff
@@ -0,0 +1,64 @@
+-- mods/default/item_entity.lua
+
+local builtin_item = minetest.registered_entities["__builtin:item"]
+
+local item = {
+	set_item = function(self, itemstring)
+		builtin_item.set_item(self, itemstring)
+
+		local stack = ItemStack(itemstring)
+		local itemdef = minetest.registered_items[stack:get_name()]
+		if itemdef then
+			self.flammable = itemdef.groups.flammable
```

Comment:
For some reason I assumed both `nil` and `0` evaluated to `false` in Lua, but it seems only `nil` does. Fixed.